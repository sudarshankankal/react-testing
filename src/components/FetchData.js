import React, { Component } from 'react'

export default class FetchData extends Component {
    constructor(props){
        super(props);

        this.state = {
            name:'alpha'
        }
    }

    handleInputChange = (e) => {
        this.setState({name:e.target.value});
    }
    render() {
        return (
            <div>
                <input type="text" value={this.state.name}/>
            </div>
        )
    }
}
