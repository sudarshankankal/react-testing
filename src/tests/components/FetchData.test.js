import { shallow } from "enzyme";
import React from "react";
import FetchData from "../../components/FetchData";
import mockAxios from "axios";

jest.mock("axios");

describe("Given FetchData Component", () => {
  test("Check State", async () => {
    let component = shallow(<FetchData />);
    let matchState = {
      name: "alpha",
      error: null,
      userId: "",
      list: [],
    };
    expect(component.state()).toEqual(matchState);
  });

  test.skip("Check Props", () => {
    let component = shallow(<FetchData />);
    expect(component.find("input").props()).toEqual({
      type: "text",
      value: "alpha",
    });
  });

  test("mock axios test", async () => {
    const component = shallow(<FetchData />);
    jest.spyOn(mockAxios, "default").mockResolvedValue({
      data: {
        userId: 111,
      },
    });
    // console.log("State Before >>",component.state());
    return component
      .instance()
      .loadData()
      .then(() => {
        // console.log("State after >>",component.state());
        expect(component.state().userId).toBe(111);
      });
  });

  test("save FetchData snapshot",() => {
      const component = shallow(<FetchData/>);
      expect(component).toMatchSnapshot();
  })

  test("check function value is truthy",() => {
    const component = shallow(<FetchData/>);
    const value = component.instance().detectUser()
    expect(value).toBeTruthy();
  })
});
