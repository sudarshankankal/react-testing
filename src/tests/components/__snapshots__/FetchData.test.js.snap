// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given FetchData Component Check FetchData snapshot 1`] = `
<ListGroup>
  <div>
    Data Not Found
  </div>
</ListGroup>
`;
